homeassistant/components/scrape/sensor.py

```diff
@@ -82,25 +83,39 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Web scrape sensor."""
-    resource_config = vol.Schema(RESOURCE_SCHEMA, extra=vol.REMOVE_EXTRA)(config)
-    rest = create_rest_data_from_config(hass, resource_config)
+    if discovery_info is None:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "moved_yaml",
+            breaks_in_ha_version="2022.12.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="moved_yaml",
+        )
+        resource_config = vol.Schema(RESOURCE_SCHEMA, extra=vol.REMOVE_EXTRA)(config)
+        rest = create_rest_data_from_config(hass, resource_config)
 
-    coordinator = ScrapeCoordinator(hass, rest, SCAN_INTERVAL)
-    await coordinator.async_refresh()
-    if coordinator.data is None:
-        raise PlatformNotReady
+        coordinator = ScrapeCoordinator(hass, rest, SCAN_INTERVAL)
+        await coordinator.async_refresh()
+        if coordinator.data is None:
+            raise PlatformNotReady
 
-    sensor_config = vol.Schema(
-        TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA
-    )(config)
+        conf = config
 
-    name: str = config[CONF_NAME]
-    unique_id: str | None = config.get(CONF_UNIQUE_ID)
+    else:
+        coordinator = discovery_info["coordinator"]
+        conf = discovery_info["config"]
 
-    select: str | None = config.get(CONF_SELECT)
-    attr: str | None = config.get(CONF_ATTRIBUTE)
-    index: int = config[CONF_INDEX]
-    value_template: Template | None = config.get(CONF_VALUE_TEMPLATE)
+    sensor_config = vol.Schema(
+        TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA
+    )(conf)
```

Comment:
The integration schema is already validated, so we should only need to re-validate it during the legacy migration (`if discovery_info is None`)

Maybe it makes sense to rename `sensor_config` to `template_config`, and rename `conf` to `parser_config`